packages/core/src/modules/ledger/IndyPool.ts

```diff
@@ -74,35 +75,39 @@ export class IndyPool {
   }
 
   public async connect() {
-    const poolName = this.poolConfig.id
-    const genesisPath = await this.getGenesisPath()
+    this.poolConnected = (async () => {
```

Comment:
Maybe we can extract this to a private method, so we don't need a self executing function? 

I think that may look a bit cleaner

Reply:
Also, maybe we should add a check in this method that we're not already connecting/connected and throw an error otherwise? Otherwise you can still call the method two times and it will try to open another connection

Reply:
Agreed, I added a check in the method that will return an error if `poolConnected` is defined. I also added a catch to `poolConnected` that will set `poolConnected` to undefined so we can retry the connection when an error occurs.

```TypeScript
  public async connect() {
    if (!this.poolConnected) {
      // Save the promise of connectToLedger to determine if we are done connecting
      this.poolConnected = this.connectToLedger()
      this.poolConnected.catch((error) => {
        // Set poolConnected to undefined so we can retry connection upon failure
        this.poolConnected = undefined
        this.logger.error('Connection to pool: ' + this.poolConfig.genesisPath + ' failed.', { error })
      })
      return this.poolConnected
    } else {
      throw new AriesFrameworkError('Cannot attempt connection to ledger, already connecting.')
    }
  }
```